src/solver/surfaceIntegrations.F90

```diff
@@ -568,22 +574,68 @@ subroutine wallIntegrationFace(localValues, mm)
             v(3) = ww2(i, j, ivz)
             v = v / (sqrt(v(1)**2 + v(2)**2 + v(3)**2) + 1e-16)
 
-            ! Dot product with free stream
-            sensor = -(v(1) * velDirFreeStream(1) + v(2) * velDirFreeStream(2) + &
-                       v(3) * velDirFreeStream(3))
+            if (sepmodel == surfvec) then
+                vectNormProd = velDirFreeStream(1) * BCData(mm)%norm(i, j, 1) + &
+                               velDirFreeStream(2) * BCData(mm)%norm(i, j, 2) + &
+                               velDirFreeStream(3) * BCData(mm)%norm(i, j, 3)
 
-            !Now run through a smooth heaviside function:
-            sensor = one / (one + exp(-2 * sepSensorSharpness * (sensor - sepSensorOffset)))
+                vectNorm(1) = velDirFreeStream(1) - vectNormProd * BCData(mm)%norm(i, j, 1)
```

Comment:
Can you add some comments here about the quantities being computed? Also, I think the variable names here could be improved. For example, `vectNorm` is not the norm of a vector or a normal vector. It seems like it is the tangential component of the freestream velocity direction?